core/store/models/job_spec_error.go

```diff
@@ -0,0 +1,25 @@
+package models
+
+import (
+	"time"
+)
+
+// JobSpecError represents an asynchronous error caused by a JobSpec
+type JobSpecError struct {
+	ID          *ID       `json:"id"`
+	JobSpecID   *ID       `json:"-"`
```

Comment:
Why is it a pointer?

Reply:
kinda just following convention here. We use a pointer for JobSpecID in `Initiator` and `TaskSpec`. Most of the functions like `orm.FindJob` expect a pointer as an argument so this makes it so we don't have to convert to and from the pointer when doing things like
```go
job, err := orm.orm.FindJob(jobSpecError.JobSpecID)
```

Reply:
Ah, I wonder if this may have something to do with gorm scan/value dumping... OK its cool